Makefile

```diff
@@ -180,8 +180,15 @@ $(BIN)/mockery: internal/tools/go.mod
 $(BIN)/enumer: internal/tools/go.mod
 	$(call go_build_tool,github.com/dmarkham/enumer)
 
-$(BIN)/goimports: internal/tools/go.mod
-	$(call go_build_tool,golang.org/x/tools/cmd/goimports)
+# it's important that the formatter is NOT whatever version they have installed,
+# so build it for this project.
+# prefer go_build_tool to support newer go versions, but either is reasonable.
+$(BIN)/gofmt: internal/tools/go.mod
+	$(call go_build_tool,cmd/gofmt)
```

Comment:
gofancyimports doesn't format, and it conflicts with goimports (which does format).

so now we have "normal" gofmt use.

perhaps we should add `-s`?

Reply:
I guess this is the issues with fancyimports. I wonder how we can use the same logic in IDEs.
Maybe we can create a wrapper that does goimports + fancyimports afterwards?

Reply:
to be clear, removing goimports and using gofmt is not a _problem_.  this is just the motivation.  gofmt is faster, and gofancyimports takes care of the imports that gofmt doesn't touch.

Reply:
ah, for IDEs: should definitely be possible, though unlike goimports it needs to be _in addition to_ gofmt, it's not a replacement.  combining those might require a small wrapper-binary 🤔 

though since it's an Analyzer, it should integrate into gopls just fine.

---

a combined-binary might be reasonable to add to gofancyimports, for IDE purposes.  GoLand has a "goimports-like" formatting option which acts a lot like this too.

Reply:
After this lands, what should we all use as Format Tool in IDE?

Reply:
Just goimports with -local afaict.  I honestly disable this though, IDE formatting does funky and incorrect things kinda frequently

Reply:
from some side chats: we'll probably bundle a formatter-wrapper to simplify vscode and maybe other tools.

that's a bit easier to discover and fix after it's merged though, so I'm just going to skip it for now.

Reply:
similar stuff with gci due to unused-import cleanup, but this does also `gofmt` so a custom wrapper to run both is probably _less_ needed - IDEs often detect and warn and offer to clean those up separately.